charts/backstage/templates/sonataflows.yaml

```
@@ -140,22 +155,41 @@ spec:
             drop:
               - ALL
         env:
+{{- if .Values.upstream.postgresql.enabled }}
         - name: PGPASSWORD
           valueFrom:
-{{- if .Values.upstream.postgresql.enabled }}
             secretKeyRef:
               name: {{ .Release.Name }}-postgresql-svcbind-postgres
               key: password
+{{- else }}
+          - name: POSTGRES_HOST
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef }}
+                key: POSTGRES_HOST
+          - name: POSTGRES_USER
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef }}
+                key: POSTGRES_USER
+          - name: POSTGRES_PORT
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef }}
+                key: POSTGRES_PORT
+          - name: POSTGRES_PASSWORD
```

```suggestion
          - name: PGPASSWORD
```

I think `PGPASSWORD` is the expected env var name.